cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/external/ExternalDataset.java

```diff
@@ -94,13 +93,4 @@ public void recordRead() {
   public void recordWrite() {
     // Nothing to do, the platform will record the access based on the @WriteOnly annotation
   }
-
-  /**
-   * Record ReadWrite access to this dataset. This method does nothing, but the
-   * @ReadWrite annotation causes recording of a read write access to this dataset.
-   */
-  @ReadWrite
-  public void recordReadWrite() {
```

Comment:
are you sure nothing is calling this through reflection? It may be safer to keep this here, though the Action code will not call it.

Reply:
This method was added by me to record READ_WRITE access using reflection. Now we only support READ and WRITE, so it is safe to remove this method